src/parser/json/snippet_parser.py

```diff
@@ -0,0 +1,128 @@
+        snippets_list = []
+        for snippet_dict in snippet_dicts_list:
+            try:
+                snippets_list.append(self.parse_snippet(snippet_dict))
+            except SPDXParsingError as err:
+                self.logger.extend(err.get_messages())
+        if self.logger.has_messages():
+            raise SPDXParsingError(self.logger.get_messages())
+
+        return snippets_list
+
+    def parse_snippet(self, snippet_dict: Dict) -> Snippet:
+        logger = Logger()
+        spdx_id: str = snippet_dict.get("SPDXID")
+        file_spdx_id: str = snippet_dict.get("snippetFromFile")
+        name: Optional[str] = snippet_dict.get("name")
+        ranges: Dict = parse_field_or_log_error(logger=logger, field=(snippet_dict.get("ranges")),
+                                                parsing_method=self.parse_ranges, default={})
+        byte_range: Tuple[int, int] = ranges.get(RangeType.BYTE)
+        line_range: Optional[Tuple[int, int]] = ranges.get(RangeType.LINE)
+        attribution_texts: List[str] = snippet_dict.get("attributionTexts")
+        comment: Optional[str] = snippet_dict.get("comment")
+        copyright_text: Optional[str] = snippet_dict.get("copyrightText")
+        license_comment: Optional[str] = snippet_dict.get("licenseComments")
+        concluded_license: Optional[Union[
+            LicenseExpression, SpdxNoAssertion, SpdxNone]] = parse_field_or_log_error(
+            logger=logger, field=snippet_dict.get("licenseConcluded"),
+            parsing_method=self.license_expression_parser.parse_license_expression, optional=True)
+
+        license_info: Optional[Union[List[
+            LicenseExpression], SpdxNoAssertion, SpdxNone]] = parse_field_or_log_error(
+            logger=logger, field=snippet_dict.get("licenseInfoInSnippets"),
+            parsing_method=self.license_expression_parser.parse_license_expression, optional=True)
+
+        if logger.has_messages():
+            raise SPDXParsingError([f"Error while parsing snippet: {logger.get_messages()}"])
+
+        snippet = construct_or_raise_parsing_error(Snippet,
+                                                   dict(spdx_id=spdx_id, name=name, byte_range=byte_range,
+                                                        file_spdx_id=file_spdx_id, line_range=line_range,
+                                                        attribution_texts=attribution_texts, comment=comment,
+                                                        copyright_text=copyright_text, license_comment=license_comment,
+                                                        concluded_license=concluded_license,
+                                                        license_info_in_snippet=license_info))
+
+        return snippet
+
+    def parse_ranges(self, ranges_from_snippet: List[Dict]) -> Dict:
+        if not ranges_from_snippet:
+            raise SPDXParsingError([f"No ranges dict provided."])
```

Comment:
Well, the json file should have a key "ranges" with a dict as value..but yes, not really error-worthy, so I delted this one and changed the error messsage below to use "snippet ranges" instead of "ranges dict".